src/Raven.Server/Smuggler/Documents/StreamSource.cs

```diff
@@ -781,7 +798,8 @@ await foreach (var reader in ReadArrayAsync())
                         reader.TryGet(nameof(SubscriptionState.LastBatchAckTime), out DateTime lastBatchAckTime) == false ||
                         reader.TryGet(nameof(SubscriptionState.LastClientConnectionTime), out DateTime lastClientConnectionTime) == false ||
                         reader.TryGet(nameof(SubscriptionState.Disabled), out bool disabled) == false ||
-                        reader.TryGet(nameof(SubscriptionState.SubscriptionId), out long subscriptionId) == false)
+                        reader.TryGet(nameof(SubscriptionState.SubscriptionId), out long subscriptionId) == false ||
+                        reader.TryGet(nameof(SubscriptionState.ArchivedDataProcessingBehavior), out ArchivedDataProcessingBehavior archivedDataProcessingBehavior) == false)
```

Comment:
why? 5.4 backups will not have this?

Reply:
Made it more 'optional' to fetch

Reply:
... (https://github.com/ravendb/ravendb/pull/16732#discussion_r1318300112) and if that's good https://github.com/ravendb/ravendb/pull/16732/commits/6b44d2f9166bd0c10b73628f9eb690cff9074dff

Reply:
we use this method [here](https://github.com/ravendb/ravendb/blob/42cf958d6e15d42b7ac63b488caca92f254ab64d/src/Raven.Server/Smuggler/Documents/SmugglerBase.cs#L1044-L1053), to get subscriptions, and then we use this [write](https://github.com/ravendb/ravendb/blob/37b3d583903a4c3cd942223d7e56220c04451d88/src/Raven.Server/Smuggler/Documents/Actions/DatabaseSubscriptionActionsBase.cs#L30-L37) on the subscriptions we've received back

by writing, we mean creating a PutCommand https://github.com/ravendb/ravendb/blob/2315e7f7b9a506e3dc92267520bcd3d5b299e70b/src/Raven.Server/Smuggler/Documents/Actions/DatabaseSaubscriptionActions.cs#L15-L23

where we don't use anything related to ArchivedDataBehavior, and I think we should
it needs to be fixed, but first, I need to know a thing

what is the flow, where we execute a put command to create a new subscription while smuggling subscriptions states? it's sharding-related?

Reply:
i've updated the part that i've been aware of, also I've prepared backward compatibility (v5.x and earlier versions subscriptions) commit